pkg/collector/check/name.go

```diff
@@ -0,0 +1,15 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2018 Datadog, Inc.
+
+package check
+
+import (
+	"strings"
+)
+
+// Name returns the name of the check extracted from its filePath
+func Name(filePath string, fileExtension string) string {
```

Comment:
I think this should live in the `providers` package (probably in the `file.go` file actually), since IMO determining a check name from its yaml file path is really specific to that provider